### ISSUE TITLE
BUG: Animation state parameter is ignored in Launch Projectiles (v3.0.3)

### ISSUE THREAD

**Issue by jserodio** [chrismaltby/gb-studio#1026]
**Describe the bug**
Animation state parameter is ignored in Launch Projectiles. It just uses the Default Idle animation of the Actor. It won't matter what Animation State you select.

**To Reproduce**
1. Create Animation State (mine is called Muerto)
2. Set frame 0 tile from Sprite Sheet.
![image](https://user-images.githubusercontent.com/5813094/151064044-d5f2cf05-3da4-4bb1-874d-66c7f4fe77fa.png)

3. Confirm that animation state works
4. Create Launch projectile
5. Set parameters: sprite sheet + new animation state in parameter
6. I used the Direction UP.
![image](https://user-images.githubusercontent.com/5813094/151063962-9a1c57e5-9031-4062-b632-65db4b2909fc.png)
7. Test launch
![testtt](https://user-images.githubusercontent.com/5813094/151065704-9539bfa7-e8a6-4a4b-9220-b5c154d1d7c4.gif)
8. Try using it attaching a button, instead of using it from a custom script, same behavior
![image](https://user-images.githubusercontent.com/5813094/151064332-489a5243-30ec-46d6-87a3-18f150dbe475.png)
![dieee](https://user-images.githubusercontent.com/5813094/151065786-55d3c6c8-6647-4aef-99f8-159de421ba15.gif)
9. In my New State, tried also changing the New State to Animation Type: Fixed Direction+ Movement (to have iddle + moving), same behaviour.
![image](https://user-images.githubusercontent.com/5813094/151064709-5654f109-a1a1-4280-a51e-ae903dbe3f34.png)
![sameshit](https://user-images.githubusercontent.com/5813094/151065938-46d3b370-b605-47c6-809e-000340be1095.gif)
10. In Default animation state, changed to Animation type: Four Directions + moving. Set Idle up to the desired frame I want to see.
11. It works, and it shows that the current animation state using is the Default, and not the new custom one.
![xxx](https://user-images.githubusercontent.com/5813094/151066187-b263c153-48b1-4fc0-b05d-3a95782401e9.gif)

**Expected behavior**
I expected the projectile to change to the frame I designed on my new animation frame.
![expected](https://user-images.githubusercontent.com/5813094/151065100-a902b409-99ef-4202-bba3-71b1b210aa99.gif)

**Screenshots**
I think we got enough. I hope provided gifs are useful.

**Platform (please complete the following information):**
 - OS: [e.g. Windows 11]
 - Version 3.0.3

**Additional context**
I also asked for help in Discord: https://discord.com/channels/554713715442712616/585023243732123676/935628891731669004

**Comment by john-lay** [comment]
I've been looking into this bug and have found that the feature was added in commit af6dcd3 and I've confirmed it was working. Another commit 8 days later bf63da8 to set the direction of the projectile breaks the functionality.

In summary, the commit to set the direction of the actor based on the direction of the actor simplifies the animation frames to up, down, left and right and uses the corresponding frame when launching the projectile.

I think this change was intentional and perhaps the ability to chose the animation state should be removed from projectiles if the bug can't be fixed.

**Comment by maxoakland** [comment]
No, the animation state setting needs to be kept because otherwise too many of our sprites will be unavailable as projectiles
